src/components/productGallery/ProductGallery.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from "react";
+import "./ProductGallery.css";
+
+const ProductGallery = ({
+  bigImage,
+  smallTopLeftImage,
+  smallTopRightImage,
+  smallBottomLeftImage,
+  smallBottomRightImage,
+}) => {
+  const [showPopupProductGallery, setShowPopupProductGallery] = useState(true);
+  const showPopupHandler = () => {
+    setShowPopupProductGallery(!showPopupProductGallery);
+  };
+
+  return (
+    <div className="product-gallery-main">
+      <div className="big-img-container">
+        <img
+          onClick={showPopupHandler}
+          src={bigImage}
+          alt="bigImage"
+          className="big-image"
+        />
+      </div>
+      <div className="small-imgs-container">
+        <img
+          onClick={showPopupHandler}
+          src={smallTopLeftImage}
+          alt="small top left image"
+          className="small-img-left"
+        />
+        <img
+          onClick={showPopupHandler}
+          src={smallTopRightImage}
+          alt="small top right image"
+          className="small-img-right-top"
+        />
+        <img
+          onClick={showPopupHandler}
+          src={smallBottomLeftImage}
+          alt="small bottom left image"
+          className="small-img-left"
+        />
+        <div className="show-photos-btn-container">
+          <img
+            onClick={showPopupHandler}
+            src={smallBottomRightImage}
+            alt="small bottom right image"
+            className="small-img-right-bottom"
+          />
+          <div className="show-photos-btn" onClick={showPopupHandler}>
+            <svg
+              xmlns="http://www.w3.org/2000/svg"
+              viewBox="0 0 16 16"
+              aria-hidden="true"
+              role="presentation"
+              focusable="false"
+              className="show-photos-btn-svg"
+            >
+              <path
+                fillRule="evenodd"
+                d="M3 11.5a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3zm5 0a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3zm5 0a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3zm-10-5a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3zm5 0a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3zm5 0a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3zm-10-5a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3zm5 0a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3zm5 0a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3z"
+              ></path>
+            </svg>
+            <span>Show all photos</span>
+          </div>
+        </div>
+      </div>
```

Comment:
You can place the conditional rendering for the popup at the end of the file like this:
{showPopupProductGallery && (
  <div className="popup-gallery">
    {/* Show popup component here */}
  </div>
)}

Reply:
It is another component. I will import it after approving this one!